server/gitrest/src/routes/summaries.ts

```diff
@@ -392,12 +392,33 @@ export function create(
         const resultP = repoManagerFactory.open(
             request.params.owner,
             request.params.repo,
-        ).then(async (repoManager) => createSummary(
-            repoManager,
-            wholeSummaryPayload,
-            documentId,
-            getExternalWriterParams(request.query?.config as string | undefined)?.enabled ?? false,
-        ));
+        ).then(async (repoManager): Promise<IWriteSummaryResponse | IWholeFlatSummary> => {
+            const externalStorageEnabled =
+                getExternalWriterParams(request.query?.config as string | undefined)?.enabled ?? false;
+            const writeSummaryResponse = await createSummary(
+                repoManager,
+                wholeSummaryPayload,
+                documentId,
+                externalStorageEnabled,
+            );
+            if (wholeSummaryPayload.type === "container") {
+                try {
+                    const wholeFlatSummary = await getSummary(
+                        repoManager,
+                        writeSummaryResponse.id,
+                        documentId,
+                        externalStorageEnabled,
+                    );
+                    return wholeFlatSummary;
+                } catch (e) {
+                    // This read is for Historian caching purposes, so it should be ignored on failure.
+                    winston.error("Failed to read latest summary after writing container summary", {
+                        documentId,
```

Comment:
Does it make sense to also log the `tenantId` here? It can also be extracted from `storageRoutingId`.